File: java/parser/src/main/java/gov/lbl/parser/parser/For_indicesVisitor.java
package gov.lbl.parser.parser;

import java.util.List;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.For_index;
import gov.lbl.parser.domain.For_indices;

import static java.util.stream.Collectors.toList;

public class For_indicesVisitor extends modelicaBaseVisitor<For_indices> {
    @Override
    public For_indices visitFor_indices(modelicaParser.For_indicesContext ctx) {
        For_indexVisitor for_indexVisitor = new For_indexVisitor();
        List<For_index> indices = ctx.for_index()
                .stream()
                .map(for_index -> for_index.accept(for_indexVisitor))
                .collect(toList());
        return new For_indices(indices);
    }
}

File: jsParser/parser/For_indicesVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const For_indices = require('../domain/For_indices')

const For_indexVisitor = require('./For_indexVisitor')

class For_indicesVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitFor_indices (ctx) {
    const indices = []

    if (ctx.for_index()) {
      const for_indexVisitor = new For_indexVisitor.For_indexVisitor()
      ctx.for_index().forEach(index => {
        indices.push(for_indexVisitor.visitFor_index(index))
      })
    }

    return new For_indices.For_indices(indices)
  }
};

For_indicesVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitFor_equation = this.visitFor_equation
exports.For_indicesVisitor = For_indicesVisitor
